src/pages/Dashboard/Versions/Detail/index.jsx

```diff
@@ -0,0 +1,471 @@
+        <div className={styles.record}>
+          <div className={styles.operator}>
+            {t(audit.role)}:&nbsp;{
+              (_.find(users, { user_id: audit.operator }) || {}).username
+            }
+          </div>
+          {this.renderReason(audit)}
+          <div className={styles.time}>{audit.status_time}</div>
+        </div>
+        <div className={styles.link}>
+          <Link to={`/dashboard/app/${version.app_id}/audits`}>
+            {t('View all records')} →
+          </Link>
+        </div>
+      </Card>
+    );
+  }
+
+  renderUpdateLog() {
+    const { appVersionStore, t } = this.props;
+    const { version, changeDescription, description } = appVersionStore;
+    const isEdit = version.status === 'draft' || version.status === 'rejected';
+
+    if (isEdit) {
+      return (
+        <div className={styles.updateLog}>
+          <p>{t('Used to describe the detailed update of this version')}:</p>
+          <textarea
+            className={styles.description}
+            value={description}
+            onChange={changeDescription}
+          />
+          <div className={styles.submitButtons}>
+            <Button
+              onClick={this.updateLog}
+              type="primary"
+              disabled={!description}
+            >
+              {t('Save')}
+            </Button>
+            <Button onClick={this.resetLog}>{t('Reset')}</Button>
+          </div>
+        </div>
+      );
+    }
+
+    return (
+      <pre className={styles.descriptionShow}>{description || t('None')}</pre>
+    );
+  }
+
+  renderConfigFile() {
+    const { appVersionStore, appStore, t } = this.props;
+    const {
+      version,
+      isLoading,
+      createError,
+      uploadError,
+      packageName,
+      checkPackageFile,
+      uploadPackage
+    } = appVersionStore;
+
+    const { appDetail } = appStore;
+    const isShowUpload = isLoading || Boolean(createError);
+    const errorFiles = _.keys(uploadError);
+
+    const isEdit = version.status === 'draft' || version.status === 'rejected';
+
+    return (
+      <Card className={styles.configFile}>
+        {!isShowUpload && (
+          <div className={styles.fileInfo}>
+            <div className={styles.name}>
+              {packageName || `${appDetail.name} ${version.name}`}
+            </div>
+            <div className={styles.time}>
+              {t('Upload time')}：{version.status_time}
+              <Link
+                className={styles.link}
+                to={`/dashboard/app/${appDetail.app_id}/deploy/${
+                  version.version_id
+                }`}
+              >
+                {t('Deploy App')}
+              </Link>
+              {isEdit && (
+                <span
+                  className={styles.link}
+                  onClick={() => this.onUploadClick()}
+                >
+                  {t('Edit')}
+                </span>
+              )}
+            </div>
+          </div>
+        )}
+        <Upload
```

Comment:
Why not combine `Upload` and `UploadShow` component, because the second is the part of `Upload`

Reply:
The upload image and package is different.